contracts/utils/structs/DoubleEndedQueue.sol

```diff
@@ -40,18 +45,19 @@ library DoubleEndedQueue {
      * data[end - 1].
      */
     struct Bytes32Deque {
-        int128 _begin;
-        int128 _end;
-        mapping(int128 => bytes32) _data;
+        uint128 _begin;
+        uint128 _end;
+        mapping(uint128 => bytes32) _data;
```

Comment:
`begin` and `end` are the "borders" of the queue. That limits the length of the queue to `2**128-1` elements

We could use `uint256` instead of `uint128`, but that would result in `_begin` and `_end` being in two different slots ... which makes many operations more expensive.

Reply:
The only SafeCast i see is in `at`. We could keep the uint128 for `_begin` and `_end`, and use a uint256 key for `_data` but we will still need `at` to do the "uint128 wrapping". 

So if the key is uint256, then we would be doing 
```
deque._data[(uint256(deque._begin) + index) % (1<<128) ]
```
or
```
deque._data[(uint256(deque._begin) + index) & type(uint128).max ]
```

We can also make the argument that:
- `length` is always returning a uint128, by construction
- `if (index >= length(deque)) revert QueueOutOfBounds();` checks that index is smaller than a length, which is a uint128, so index is a uint128
- we can do a direct cast.

Reply:
That makes sense. I didn't realize before that the length was always uint128, since the code used to say "we assume there are at most int256.max items in the queue", but apparently it was already provably uint128 at most.

Let's do that and remove the safe cast.

Reply:
done